src/transformers/modeling_utils.py

```diff
@@ -1777,10 +1789,29 @@ def save_pretrained(
 
         # Save the config
         if is_main_process:
-            model_to_save.config.save_pretrained(save_directory)
+            if not getattr(model_to_save, "_hf_peft_config_loaded", False):
```

Comment:
Just note that this attribute is accessed multiple times. So if using `getattr` is preferred, I would suggest to assign the attribute to a variable and use that throughout.